CTX.js

```diff
@@ -38,129 +42,54 @@
  * This format is used in several places online, including Brown University's FreeCite
  * Citation parser (http://freecite.library.brown.edu/welcome) and Oslo University's
  * X-Port (http://www.ub.uio.no/portal/gs.htm or http://x-port.uio.no/).
- 
- * Our input looks like this:
-<ctx:context-objects xmlns:xsi='http://www.w3.org/2001/XMLSchema-instance' xsi:schemaLocation='info:ofi/fmt:xml:xsd:ctx http://www.openurl.info/registry/docs/info:ofi/fmt:xml:xsd:ctx' xmlns:ctx='info:ofi/fmt:xml:xsd:ctx'>
-<ctx:context-object timestamp='2010-01-02T16:55:48-05:00' encoding='info:ofi/enc:UTF-8' version='Z39.88-2004' identifier=''>
- <ctx:referent>
-  <ctx:metadata-by-val>
-   <ctx:format>info:ofi/fmt:xml:xsd:journal</ctx:format>
-   <ctx:metadata>
-	<journal xmlns:rft='info:ofi/fmt:xml:xsd:journal' xsi:schemaLocation='info:ofi/fmt:xml:xsd:journal http://www.openurl.info/registry/docs/info:ofi/fmt:xml:xsd:journal'>
-	 <rft:atitle>Acute Myocardial Infarction in the Medicare population: process of care and clinical outcomes</rft:atitle>
-	 <rft:spage>2530</rft:spage>
-	 <rft:date>1992</rft:date>
-	 <rft:stitle>Journal of the American Medical Association</rft:stitle>
-	 <rft:genre>article</rft:genre>
-	 <rft:volume>18</rft:volume>
-	 <rft:epage>2536</rft:epage>
-	 <rft:au>I S Udvarhelyi</rft:au>
-	 <rft:au>C A Gatsonis</rft:au>
-	 <rft:au>A M Epstein</rft:au>
-	 <rft:au>C L Pashos</rft:au>
-	 <rft:au>J P Newhouse</rft:au>
-	 <rft:au>B J McNeil</rft:au>
-	</journal>
-   </ctx:metadata>
-  </ctx:metadata-by-val>
- </ctx:referent>
-</ctx:context-object>
-</ctx:context-objects>
  *
  * The approach we will take is to convert this into COinS, so that we can
  * piggy-back off of the perhaps more robust support in the core Zotero code.
  */
 
-function detectWeb(doc, url) {
-	var texts = [], text = "";
-	var codes = doc.getElementsByTagName("code");
-	for(var i = 0; i < codes.length; i++) {
-		text = codes[i].textContent;
-		text.replace(/</g,"&lt;").replace(/>/g,"&gt;");
-		texts.push(text);
-	}
-	return detectInString(texts);
-};
-
-function doWeb(doc, url) {
-	var texts = [], text = "";
-	var codes = doc.getElementsByTagName("code");
-	for(var i = 0; i < codes.length; i++) {
-		text = codes[i].textContent;
-		text.replace(/</g,"&lt;").replace(/>/g,"&gt;");
-		texts.push(text);
-	}
-	doImportFromText(texts, true);
-};
-
-function doImport() {
-	var text = "";
-	var line;
-	while(line = Zotero.read()) {
-		text += line;
-	}
-	return doImportFromText(text, false);
-}
-
 function detectImport() {
-	var text = "";
-	var line;
-	while(line = Zotero.read()) {
-		text += line;
-	}	
-	return detectInString(text) != false;
+	var doc = Zotero.getXML().documentElement;
```

Comment:
Let's avoid this in detect until we need to make sure that the XML is valid. We wouldn't want to be parsing large XML files and loading them into memory for no reason. I think if you stick to your previous strategy of reading ~100 lines and looking for a signature string and only then calling `Zotero.getXML` if the string is found, that would be ideal.